packages/build/src/lib/plugins/angular-rspack-plugin.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  Compiler,
+  Compilation,
+  sources,
+  RspackPluginInstance,
+  RspackOptionsNormalized,
+} from '@rspack/core';
+import {
+  AngularRspackPluginOptions,
+  normalizeOptions,
+  NG_RSPACK_SYMBOL_NAME,
+  NgRspackCompilation,
+} from '../models';
+import { JavaScriptTransformer } from '@angular/build/src/tools/esbuild/javascript-transformer';
+import { maxWorkers } from '../utils/utils';
+import {
+  setupCompilationWithParallelCompilation,
+  buildAndAnalyzeWithParallelCompilation,
+} from '@ng-rspack/compiler';
+import { dirname, normalize, resolve } from 'path';
+import {
+  Hash,
+  MapOptions,
+  RawSourceMap,
+} from '@rspack/core/compiled/webpack-sources';
+
+const PLUGIN_NAME = 'AngularRspackPlugin';
+type Awaited<T> = T extends Promise<infer U> ? U : T;
+type ResolvedJavascriptTransformer = Parameters<
+  typeof buildAndAnalyzeWithParallelCompilation
+>[2];
+
+export class AngularRspackPlugin implements RspackPluginInstance {
+  #_options: AngularRspackPluginOptions;
+  #typescriptFileCache: Map<string, string>;
+  #javascriptTransformer: ResolvedJavascriptTransformer;
+  // This will be defined in the apply method correctly
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-expect-error
+  #angularCompilation: Awaited<
+    ReturnType<typeof setupCompilationWithParallelCompilation>
+  >;
+
+  constructor(options: Partial<AngularRspackPluginOptions>) {
+    this.#_options = normalizeOptions(options);
```

Comment:
Exactly what i thought last time testing related code. The typing has no good orga of where it is open and where closed. We could scan the code and move other normalization logic also into `normalizeOptions`.
But I guess best when the tests are covering existing logic.